ydb/docs/en/core/deploy/configuration/config.md

```diff
@@ -456,6 +456,109 @@ actor_system_config:
+
+Examples of components managed by the memory controller:
+
+- [Shared cache](../../concepts/glossary.md#shared-cache): stores recently accessed data pages read from Blob Storage to reduce disk I/O and accelerate data retrieval.
+- [MemTable](../../concepts/glossary.md#memtable): holds data that has not yet been flushed to SST.
+- [KQP](../../concepts/glossary.md#kqp): stores intermediate query results.
+- Allocator Caches: keeps memory blocks which have been released but not yet returned to the operating system.
+
+Memory limits can be configured to control overall memory usage, ensuring the database operates efficiently within the available resources.
+
+### Hard memory limit {#hard-memory-limit}
+
+The hard memory limit specifies the total amount of available memory.
+
+By default, the hard memory limit is set to the {{ ydb-short-name }} node process's [cgroups](https://en.wikipedia.org/wiki/Cgroups) memory limit.
+
+In environments without a cgroups memory limit, the default hard memory limit is equal to the host's total available memory. This allows the database to utilize all available resources in unrestricted environments, though it may lead to resource contention with other processes running on the same host.
+
+Additionally, the hard memory limit can be specified in the configuration. Note that the database process may still exceed this limit, so it is highly recommended to use cgroups memory limits in production environments to enforce strict memory control.
+
+Example of the `memory_controller_config` section with a specified hard memory limit:
+
+```yaml
+memory_controller_config:
+  hard_limit_bytes: 16106127360
+```
+
+### Soft memory limit {#soft-memory-limit}
+
+The soft memory limit specifies a dangerous threshold that should not be exceeded under normal conditions.
+
+If the soft limit is exceeded, {{ ydb-short-name }} starts to reduce the shared cache size to zero. Therefore, more database nodes should be added to the cluster as soon as possible or per-component memory limits should be reduced.
+
+### Target memory utilization {#target-memory-utilization}
+
+The target memory utilization specifies a threshold for memory usage that is considered optimal.
+
+Flexible cache sizes are calculated according to their limit thresholds to keep process consumption around this value.
+
+For example, in a database that consumes a little memory on query execution, caches consume memory around this threshold and other memory stays free. And if query execution consumes more memory, caches start to reduce their limits to their minimum threshold.
```

Comment:
```suggestion
### Target memory utilization {#target-memory-utilization}

The target memory utilization specifies a threshold for memory usage that is considered optimal.

Flexible cache sizes are calculated according to their limit thresholds to keep process consumption around this value.

For example, in a database that consumes little memory on query execution, caches consume memory around this threshold, and other memory stays free. If query execution consumes more memory, caches start to reduce their limits to their minimum threshold.
```

Reply:
Why `consumes little memory`?

Little usually refers to 'not enough'